packages/server/lib/modes/record.js

```diff
@@ -209,10 +209,19 @@ const uploadArtifacts = (options = {}) => {
   }
 
   if (captureUploadUrl && protocolManager) {
+    const successCallback = success('Test Replay', captureUploadUrl, { key: 'protocol', statFile: false })
```

Comment:
any reason we define these above array insertion? We are at least defining one callback per upload that isn't going to be used

Reply:
Mainly it was so I could DRY up the failCallback since I was using it twice in separate callbacks.

Reply:
If you re-throw `res.error` in the `.then` block, I believe the `.catch` will handle it